### PR TITLE
fix: boolean operators on boolean operands

### PR DESCRIPTION
**Pull request by priyasiddharth**
Also innocuous clang format changes

### CODE REVIEW

**Comment by priyasiddharth** [comment]
@jorgenavas can you PTAL? I found these through compiler warnings.

**Comment by caballa** [comment]
Can you pinpoint the warning fixes? I cannot tell with clang-format.
I want to double check because most of the warning fixes are in the `dev` branch which I'm planning to merge with master today or tomorrow.

File: include/crab/domains/region_domain.hpp [hunk 1]
```diff
@@ -757,7 +753,7 @@ class region_domain final : public abstract_domain_api<region_domain<Params>> {
       return;
     } else if (o.is_bottom()) {
       return;
-    } else if (is_top() | o.is_top()) {
+    } else if (is_top() || o.is_top()) {
```
Comment (priyasiddharth) [inline include/crab/domains/region_domain.hpp:756]: this is important

File: include/crab/domains/region_domain.hpp [hunk 2]
```diff
@@ -814,7 +810,7 @@ class region_domain final : public abstract_domain_api<region_domain<Params>> {
       return o;
     } else if (o.is_bottom()) {
       return *this;
-    } else if (is_top() | o.is_top()) {
+    } else if (is_top() || o.is_top()) {
```
Comment (priyasiddharth) [inline include/crab/domains/region_domain.hpp:813]: this is important

File: include/crab/domains/union_find_domain.hpp [hunk 3]
```diff
@@ -207,82 +198,82 @@ class union_find_domain {
...
     crab::CrabStats::count("union_find.count.join_or_widening");
     crab::ScopedCrabStats __st__("union_find.join_or_widening");
-    
+
     if (is_bottom()) {
       return o;
     } else if (o.is_bottom()) {
       return *this;
-    } else if (is_top() | o.is_top()) {
+    } else if (is_top() || o.is_top()) {
```
Comment (priyasiddharth) [inline include/crab/domains/union_find_domain.hpp:217]: this is important

Comment (caballa) [reply include/crab/domains/union_find_domain.hpp:217]: Thanks @priyasiddharth! . I'll do the fixes by myself to separate all the noise from clang-format.

**Comment by caballa** [comment]
Done in commit [e4841ed](https://github.com/seahorn/crab/commit/e4841edf587396e9668a3113f2dd3e2b895fe705)

**Pull request closed without merging**